src/Checkbox/Checkbox.styles.ts

```diff
@@ -0,0 +1,30 @@
+import styled from 'styled-components';
+import { CheckboxProps } from './types';
+import { getBackgroundColorFromState, getBorderColorFromState, getColorFromState } from './utils';
+
+interface StyledCheckboxProps extends CheckboxProps {
+  $indeterminate?: boolean;
+}
+
+export const StyledCheckbox = styled.input.attrs<StyledCheckboxProps>({ type: 'checkbox' })<StyledCheckboxProps>`
```

Comment:
If you want you can add the type checkbox in the component. Just for not having some props here and another in the component.